collections/_posts/2023-02-27-ConvNextV2.md

```diff
@@ -0,0 +1,138 @@
+
+* ConvNeXt V1 was focus on supervised learning only however, similarly to transformers, convolution based models can benefit from self-supervised learning techniques (such as MAE)
+* Simply apply self-supervised learning methods (i.e. MAE) to ConvNeXt leads to sub optimal results
+* **This article propose a fully convolutional masked autoencoder framework (MAE) and modify the ConvNeXt architecture with Global Response Normalization (GRN) layers**
+
+
+
+# Fully convolutional masked autoencoder
+
+![](/collections/images/convnextV2/framework.jpg) 
+
+* They use a random masking strategy with a masking ratio of 0.6 
+
+* They use ConvNeXt as the encoder
+
+* Unlike transformers you can't simply remove masked patches from the image as the 2D structure of the image must be preserved for convolution
+
+* Also naives solutions such as replace masked patches by *masked tokens* doesn't perform well in practice
+
+* The idea here is to see the masked image as a sparse data, based on that is natural to use sparse convolution that will operate only on the visible pixels
+  
+  | *Note 1 : if the center pixel of the convolution is masked then the convolution will not operate and just return a masked pixel*
+
+  |  *Note 2 :  the sparse convolution layers can be converted back to standard convolution at the fine-tuning stage without requiring additional handling*
+  
+* They tested several decoder architecture/depth but at the end they choose a simple ConvNeXt plain decoder 
+
+* Loss function is the mean squared error (MSE) computed only on the masked patches between reconstructed and target images
+  
+  ## Evaluation of FCMAE
+
+* They pre-train and fine-tune on ImageNet-1K for 800 and 100 epochs respectively
+
+* Ablation study is done to justify design choice
+
+  ![](/collections/images/convnextV2/sparse_conv.jpg)
+
+  ![](/collections/images/convnextV2/ablation1.jpg)
+
+* The also compare the self-supervised approach to fully supervised learning
+
+  <img src="/collections/images/convnextV2/result1.jpg" style="zoom:70%;" />
+
+* **They perform better than the fully supervised setup on 100 epochs but are still worse than the  orginal ConvNeXt V1 baseline train on 300 epochs**
+
+  | *This is in contrast to the recent success of masked image modeling using transformer-based models [..] where the pre-trained models significantly outperform the supervised counterparts.*
+
+# Global Response Normalization
+
+- To try to improve on that and to gain more insight into the learning behavior they perform a qualitative analysis in the feature space.
+
+<img src="/collections/images/convnextV2/feature_collapse.jpg" style="zoom:100%;" />
+
+- They noticed a feature collapse phenomenom and they computed the cosine distance between features to get more insight
+
+  <img src="/collections/images/convnextV2/cosine_distance.jpg" style="zoom:100%;" />
+
+  
+
+- This analysis showed a reduction in feature diversity through the network for the ConvNeXt V1 FCMAE model
```

Comment:
It might be useful to quickly recall what is the difference between V1 FCMAE and V2 FCMAE since it is not written explicitely before.